Smoothed out the movement play bar while using a MediaElement backend

jj227 opened the pull request:
As mentioned in #498, #592, and #611 

I smoothed out the movement of the play bar while using a MediaElement backend by polling for updates using requestAnimationFrame.  The update loop is broken on pause and ended.

You can see the improvement by comparing:
http://rawgit.com/jj227/wavesurfer.js/master/example/annotation/index.html
http://rawgit.com/katspaugh/wavesurfer.js/master/example/annotation/index.html

katspaugh commented on src/mediaelement.js:
```diff
@@ -105,6 +106,14 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
+                requestAnimationFrame(updateProgress);
+            }
+        }
+        this.updateProgressAnimating = requestAnimationFrame(updateProgress);
```
It shouldn't be part of the drawer. It should be used in the drawer code. I'm OK for having `requestAnimationFrame` for both Web Audio and Media Element.

katspaugh replied:
Sorry, not in the drawer code, but in `wavesurfer.js` (since it has access to both the backend and the drawer).

katspaugh commented on src/mediaelement.js:
```diff
@@ -105,6 +106,14 @@ WaveSurfer.util.extend(WaveSurfer.MediaElement, {
+        function updateProgress(){
+            my.fireEvent('audioprocess', my.getCurrentTime());
+            if (my.updateProgressAnimating) {
+                requestAnimationFrame(updateProgress);
```
Doesn't it still need vendor prefixes in some browsers?

katspaugh commented:
We used to have `requestAnimationFrame`:
https://github.com/katspaugh/wavesurfer.js/commit/786ac62d8d8f3ff9cc91db84edb860c4c30ea6cf

The reason was that I thought it would solve #379, but now I realize `requestAnimationFrame` had nothing to do with that problem.

jj227 commented:
the old requestAnimationFrame code you had was a little more concise with how it stops the loop so i merged that back in with two small changes:

- i added in a firing of the audioprocess event because my use case involves keeping a separate play bar in sync using that event.  i suppose i could use two animation loops to keep the updated separately if you don't think this is a good idea.
- most browsers stopped using the prefixes mid-2013 but i left in the webkit prefix and added a second prefix for old versions of firefox

katspaugh commented:
> i added in a firing of the audioprocess event

No harm in that, I guess.

> most browsers stopped using the prefixes for requestAnimationFrame mid-2013

Cool! Good to know.

Can you [squash](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/) all commits into one, please? You can do that with `git rebase -i HEAD~13`.

Thanks!

katspaugh commented:
Thank you! :thumbsup:
